packages/trie/src/proof/range.ts

```diff
@@ -443,28 +446,28 @@ export async function verifyRangeProof(
     return false
   }
 
-  if (proof === null || firstKey === null || lastKey === null) {
-    throw new Error(
-      'invalid all elements proof: proof, firstKey, lastKey must be null at the same time'
-    )
-  }
-
   // Zero element proof
   if (keys.length === 0) {
     const { trie, value } = await verifyProof(
       rootHash,
-      nibblestoBytes(firstKey),
-      proof,
+      nibblestoBytes(firstKey as any),
+      proof as any,
       useKeyHashingFunction
     )
 
-    if (value !== null || (await hasRightElement(trie, firstKey))) {
+    if (value !== null || (await hasRightElement(trie, firstKey as any))) {
       throw new Error('invalid zero element proof: value mismatch')
     }
 
     return false
   }
 
+  if (proof === null || firstKey === null || lastKey === null) {
+    throw new Error(
+      'invalid all elements proof: proof, firstKey, lastKey must be null at the same time'
```

Comment:
This is new, but has to do with an all elements proof :thinking: (looks correct to me, but maybe add a test :smile: )

Reply:
This check is actually not new, I've only moved it bellow where zero-element proof checking happens. It was previously like this:
```
  if (proof === null || firstKey === null || lastKey === null) {
    throw new Error(
      'invalid all elements proof: proof, firstKey, lastKey must be null at the same time'
    )
  }

  // Zero element proof
  if (keys.length === 0) {
    const { trie, value } = await verifyProof(
      rootHash,
      nibblestoBytes(firstKey),
      proof,
      useKeyHashingFunction
    )

    if (value !== null || (await hasRightElement(trie, firstKey))) {
      throw new Error('invalid zero element proof: value mismatch')
    }

    return false
  }
```
The reason I did that was because zero-element proof checking was failing when I was passing in just the `root`, `firstKey`, and `proof`, with `null` for `lastKey` and empty arrays for `keys` and `values`, as such:
```
          const isMissingRightRange = await Trie.verifyRangeProof(
            this.root,
            origin,
            null,
            [],
            [],
            <any>rangeResult.proof,
            { useKeyHashingFunction: keccak256 }
          )
```
The error resulting from this was confusing a zero-element request to be a failed all-elements request:
```
2024-02-16T00:22:13.531Z client:AccountFetcher Error: invalid all elements proof: proof, firstKey, lastKey must be null at the same time
    at Module.verifyRangeProof (/Users/indigoalpha/code/ethjs/0/ethereumjs-monorepo/packages/trie/src/proof/range.ts:450:11)
    at Function.verifyRangeProof (/Users/indigoalpha/code/ethjs/0/ethereumjs-monorepo/packages/trie/src/trie.ts:208:12)
    at AccountFetcher.request (/Users/indigoalpha/code/ethjs/0/ethereumjs-monorepo/packages/client/src/sync/fetcher/accountfetcher.ts:400:50)
    at /Users/indigoalpha/code/ethjs/0/ethereumjs-monorepo/packages/client/test/sync/fetcher/accountfetcher.spec.ts:313:17
    at runTest (file:///Users/indigoalpha/code/ethjs/0/ethereumjs-monorepo/node_modules/@vitest/runner/dist/index.js:486:11)
    at runSuite (file:///Users/indigoalpha/code/ethjs/0/ethereumjs-monorepo/node_modules/@vitest/runner/dist/index.js:594:15)
    at runSuite (file:///Users/indigoalpha/code/ethjs/0/ethereumjs-monorepo/node_modules/@vitest/runner/dist/index.js:594:15)
    at runFiles (file:///Users/indigoalpha/code/ethjs/0/ethereumjs-monorepo/node_modules/@vitest/runner/dist/index.js:645:5)
    at startTests (file:///Users/indigoalpha/code/ethjs/0/ethereumjs-monorepo/node_modules/@vitest/runner/dist/index.js:654:3)
    at file:///Users/indigoalpha/code/ethjs/0/ethereumjs-monorepo/node_modules/vitest/dist/entry.js:278:7
```